packages/block-editor/src/components/simulate-media-query/index.js

```diff
@@ -74,6 +74,10 @@ export default function useSimulatedMediaQuery( marker, width ) {
 				++ruleIndex
 			) {
 				const rule = styleSheet.cssRules[ ruleIndex ];
+				if ( rule.type !== window.CSSRule.STYLE_RULE ) {
```

Comment:
Elsewhere in this file, we guard against a falsey `window` object. I'm not entirely sure if we need to, and those checks are causing at least one issue unrelated to this pull request (https://github.com/WordPress/gutenberg/pull/19082#discussion_r379210977).

Reply:
Oooh, I see. We could return an empty array instead of returning nothing.

Reply:
> Oooh, I see. We could return an empty array instead of returning nothing.

On further look, I'm not sure we even need to handle it. In other words, this condition isn't necessary:

https://github.com/WordPress/gutenberg/blob/56ff721e76b6412d45eae43df10dcf6524b4f83b/packages/block-editor/src/components/simulate-media-query/index.js#L18-L20

Because,

1. If `window` were _undeclared_, this would still be throwing.

Example:

```js
if ( ! someValueThatDoesntExist ) {}
// VM184:1 Uncaught ReferenceError: someValueThatDoesntExist is not defined
//    at <anonymous>:1:1
```

Testing undeclared would be as: `if ( typeof window !== 'undefined' ) {`.

Thus, we assume that if we don't need to check for it being undeclared, we must then be anticipating on of possible [falsey values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy).

2. The behavior of `get` will result in the fallback empty array for any falsey type.

https://github.com/WordPress/gutenberg/blob/56ff721e76b6412d45eae43df10dcf6524b4f83b/packages/block-editor/src/components/simulate-media-query/index.js#L22

```js
n_ > _.get( false, [ 'document', 'styleSheets' ], [] );
[]
n_ > _.get( 0, [ 'document', 'styleSheets' ], [] );
[]
n_ > _.get( 0n, [ 'document', 'styleSheets' ], [] );
[]
n_ > _.get( '', [ 'document', 'styleSheets' ], [] );
[]
n_ > _.get( null, [ 'document', 'styleSheets' ], [] );
[]
n_ > _.get( undefined, [ 'document', 'styleSheets' ], [] );
[]
n_ > _.get( NaN, [ 'document', 'styleSheets' ], [] );
[]
```

3. Thus, while you're right that we would want to return an empty array, we shouldn't need to explicitly handle the presence of `window`, since the result from Lodash `filter` will be the same regardless.

Either that, or we need to be testing that `window` is undeclared and improve the condition 😅